include/core/mir/mutex.h

```diff
@@ -33,11 +33,18 @@ class MutexGuard
 {
 public:
     MutexGuard(std::unique_lock<std::mutex>&& lock, Guarded& value)
-        : value{value},
+        : value{&value},
           lock{std::move(lock)}
     {
     }
-    MutexGuard(MutexGuard&& from) = default;
+
+    MutexGuard(MutexGuard&& from) noexcept
+        : value{from.value},
+          lock{std::move(lock)}
+    {
+        from.value = nullptr;
+    }
+
     ~MutexGuard() noexcept(false)
     {
         if (lock.owns_lock())
```

Comment:
pre-existing, but why do we need to explicitly unlock the lock on destruction? Doesn't `unique_lock` already do that for us?